### PR TITLE
Fix using system user for contact in some places

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh/issues/6302

### CODE REVIEW

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java [hunk 1]
```diff
@@ -73,7 +77,8 @@ default int getC_BP_Location_ID(final I_M_ShipmentSchedule sched)
 
 	I_AD_User getAD_User(I_M_ShipmentSchedule sched);
 
-	int getAD_User_ID(I_M_ShipmentSchedule sched);
+	@Nullable
+	BPartnerContactId getADUserID(@NonNull I_M_ShipmentSchedule sched);
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java:81]: IMHO u can call it `getBPartnerContactId`

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java [hunk 2]
```diff
@@ -73,7 +77,8 @@ default int getC_BP_Location_ID(final I_M_ShipmentSchedule sched)
 
 	I_AD_User getAD_User(I_M_ShipmentSchedule sched);
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java:78]: mark this one deprecated. advice the dev to use  `getBPartnerContactId`

Comment (teosarca) [reply de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java:78]: also pls rename it to `getBPartnerContact`, for consistency

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java (hunk 1 above)
Comment (TheBestPessimist) [reply de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java:81]: done

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java (hunk 2 above)
Comment (TheBestPessimist) [reply de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java]: done

File: de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/spi/impl/DocumentPrintingQueueHandler.java [hunk 3]
```diff
@@ -97,7 +97,7 @@ public void afterEnqueueBeforeSave(final I_C_Printing_Queue queueItem, final I_A
 		{
 			logger.debug("Setting column of C_Printing_Queue {} from DocAction-PO {} that is referenced by AD_Archive {}: [AD_User_ID={}]",
 					new Object[] { queueItem, archiveRerencedModel, archive, document.getDoc_User_ID() });
-			queueItem.setAD_User_ID(document.getDoc_User_ID());
+			queueItem.setAD_User_ID(document.getDoc_User_ID());  // TODO tbp: not sure if here i should use BPartnerContactId
```
Comment (metas-ts) [inline de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/spi/impl/DocumentPrintingQueueHandler.java:100]: neither am I, so let's stick with how you changed it here

**Pull request merged**